### ISSUE TITLE
Enable Default DB 

### ISSUE THREAD

**Issue by virmundi** [arangodb/arangodb-java-driver#109]
I'm the maintainer of [migrantverde](https://github.com/deusdat/migrantverde). I'm in the process of updating the code to support the latest driver. Please add a feature to set the default database to something other than _system.  Most operations for a migration (and probably for a normal application use cases) focus on modifying a single database, or as we say in SQL a schema. Having a default makes many operations simpler. For example, creating an index does not require carrying around the database name through all the layer. It would also save on wrapping the Arango components with a class that carriers around a default database schema and the driver itself.

**Comment by mpv1989** [comment]
Hi @virmundi 

the idea behind the new API is a clearer separation of the operations done on system/database/collection level. A default database or a switchable default database doesn't fit into this architecture.

The intended use is to create a instance of ```ArangoDB``` on your top layer and for underlying layers - which only works on database level - you create an instance of ```ArangoDatabase``` over the method ```ArangoDB.db(String)``` and pass it as a parameter into that layer.  Further layers - which only works on collection level - get an instance of ```ArangoCollection``` created from ```ArangoDatabase.collection(String)```.

```java
ArangoDB arangoDB = new ArangoDB.Builder().build();
ArangoDatabase database = arangoDB.db("myDatabase"); 
doOnDatabase(database);

public void doOnDatabase(ArangoDatabase database) {
  ArangoCollection collection = database.collection("myCollection");
  doOnCollection(collection);
}

public void doOnCollection(ArangoCollection collection) {
  ...
}
```

But I also understand your point of view, so I implemented some convenience methods today which may be helpful for you. It is now possible to get the "parent" level from the classes ```ArangoCollection``` and ```ArangoDatabase```. It is currently only available with version `4.1.11-SNAPSHOT`.

```java
public void doOnCollection(ArangoCollection collection) {
  ArangoDatabase database = collection.db(); 
  ArangoDB arangoDB = database.arango();
}
```

I think this is a good compromise. I appreciate any feedback.

**Comment by virmundi** [comment]
I agree. That's a good compromise. I look forward to using it.
